src/charm.py

```diff
@@ -24,22 +28,80 @@ def __init__(self, *args: typing.Any) -> None:
             args: passthrough to CharmBase.
         """
         super().__init__(*args)
+        self._charm_state = CharmState(charm_config=self.config)
+        self._webserver = GunicornWebserver(charm_state=self._charm_state)
         self.framework.observe(self.on.config_changed, self.config_service)
```

Comment:
Can we rename the handler `_on_config_changed` as per specs ?
https://docs.google.com/document/d/1G62PosrObvmQY5KbxvqaxByojlhDxrmNtcbPS39YbaY/edit#bookmark=id.20k0qpk5irc9

Reply:
Updated, thanks!